data_graph/services.py

```diff
@@ -0,0 +1,104 @@
+    BY_DEPARTMENT = 0
+    BY_STAFF_TITLE = 1
+    BY_AGE_DISTRIBUTION = 2
+
+
+class TrainHoursWorkloadGroupingType():
+    '''provide graph params TrainHoursWorkloadGroupingType enum'''
+    BY_TOTAL_STAFF_NUM = 0
+    BY_TOTAL_TRAINING_HOURS = 1
+    BY_PER_CAPITA_TRAINING_HOURS = 2
+    BY_TOTAL_WORKLOAD = 3
+    BY_PER_CAPITA_WORKLOAD = 4
+
+
+class DataGraphService:
+    '''Provide services for selecting data-get services'''
+
+    @staticmethod
+    def select_sub_service(request_data):
+        '''to call a specific service for getting data'''
+
+
+class DataGraphParamsService:
+    '''provide services for getting all graph params'''
+    staff_grouping_type = [
+        {'type': StaffGroupingType.BY_DEPARTMENT, 'name': '按学院'},
+        {'type': StaffGroupingType.BY_STAFF_TYPE, 'name': '按人员类别'},
+        {'type': StaffGroupingType.BY_STAFF_TITLE, 'name': '按职称'},
+        {'type': StaffGroupingType.BY_HIGHEST_DEGREE, 'name': '按最高学位'},
+        {'type': StaffGroupingType.BY_AGE_DISTRIBUTION, 'name': '按年龄分布'}
+    ]
+    trainee_number_coverage_grouping_type = [
+        {
+            'type': TraineeNumberCoverageGroupingType.BY_DEPARTMENT,
+            'name': '按学院'
+        },
+        {
+            'type': TraineeNumberCoverageGroupingType.BY_STAFF_TITLE,
+            'name': '按职称'
+        },
+        {
+            'type': TraineeNumberCoverageGroupingType.BY_AGE_DISTRIBUTION,
+            'name': '按年龄分布'
+        }
+    ]
+    training_hours_workload_grouping_type = [
+        {'type': TrainHoursWorkloadGroupingType.BY_TOTAL_STAFF_NUM,
+         'name': '按总人数'},
+        {'type': TrainHoursWorkloadGroupingType.BY_TOTAL_TRAINING_HOURS,
+         'name': '按总培训学时'},
+        {'type': TrainHoursWorkloadGroupingType.BY_PER_CAPITA_TRAINING_HOURS,
+         'name': '按人均培训学时'},
+        {'type': TrainHoursWorkloadGroupingType.BY_TOTAL_WORKLOAD,
+         'name': '按总工作量'},
+        {'type': TrainHoursWorkloadGroupingType.BY_PER_CAPITA_WORKLOAD,
+         'name': '按人均工作量'}
+    ]
+    statistics_type = [
+        {'type': StatisticsType.STAFF_STATISTICS,
+         'option': {'name': '教职工人数统计', 'subOption': staff_grouping_type}},
+        {'type': StatisticsType.TRAINEE_STATISTICS,
+         'option': {'name': '培训人数统计',
+                    'subOption': trainee_number_coverage_grouping_type}},
+        {'type': StatisticsType.FULL_TIME_TEACHER_TRAINED_COVERAGE,
+         'option': {'name': '专任教师培训覆盖率统计',
+                    'subOption': trainee_number_coverage_grouping_type}},
+        {'type': StatisticsType.TRAINING_HOURS_WORKLOAD_STATISTICS,
+         'option': {'name': '培训学时与工作量统计',
+                    'subOption': training_hours_workload_grouping_type}}
+    ]
+
+    @classmethod
+    def get_graph_param(cls):
+        '''return a data graph select dictionary'''
+        departments = Department.objects.all()
+        departments = [department.name for department in departments]
```

Comment:
If you only care about the name field of department instances, then there is no need to retrieve all fields but use one only. Suggestion:
```python
>>> Department.objects.all().values_list('name', flat=True)
<QuerySet ['人文与社会科学学部', '创新创业学院', ...]>
```

BTW, I think you should request the server with department id, instead of department name, so you might have to return department ids as well.